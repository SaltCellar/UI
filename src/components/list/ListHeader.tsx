import { css } from '@patternfly/react-styles';
import React, { ReactNode } from 'react';
import { styles } from './header.styles';

interface ListHeaderProps {
  children: ReactNode;
}

export const ListHeader: React.SFC<ListHeaderProps> = ({ children }) => (
  <header className={css(styles.header)}>{children}</header>
);
