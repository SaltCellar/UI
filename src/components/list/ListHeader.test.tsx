import { Title } from '@patternfly/react-core';
import * as emotion from 'emotion';
import { shallow } from 'enzyme';
import { createSerializer } from 'jest-emotion';
import React from 'react';
import { ListHeader } from './ListHeader';

expect.addSnapshotSerializer(createSerializer(emotion));

test('render list header', () => {
  const view = shallow(
    <ListHeader>
      <h1>Header Title</h1>
      <small>small description</small>
    </ListHeader>
  );
  expect(view).toMatchSnapshot();
});
