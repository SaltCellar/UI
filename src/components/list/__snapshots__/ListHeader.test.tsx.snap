// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render list header 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: var(--pf-global--spacer--xl);
  background-color: var(--pf-global--BackgroundColor--100);
}

<header
  className="emotion-0"
>
  <h1>
    Header Title
  </h1>
  <small>
    small description
  </small>
</header>
`;
